lib/reek/context/code_context.rb

```diff
@@ -95,16 +98,12 @@ def each(&block)
 
       alias_method :parent, :context
 
-      # Register a child context. The child's parent context should be equal to
-      # the current context.
-      #
-      # This makes the current context responsible for setting the child's
-      # visibility.
+      # Register a child context. Used by a child context to register itself with its parent.
```

Comment:
But now the comment and the method name have diverged. 
Comment says "used by a child context to register itself".
Method name says "append_child_context". Shouldn't it then be "register_self"? And why do we pass in the "child"?
This is super-confusing.